src/EditorFeatures/Core.Wpf/QuickInfo/OnTheFlyDocsView.xaml.cs

```diff
@@ -135,31 +179,61 @@ private async Task SetResultTextAsync(ICopilotCodeAnalysisService copilotService
 
         try
         {
-            var response = await copilotService.GetOnTheFlyDocsAsync(_onTheFlyDocsInfo.SymbolSignature, _onTheFlyDocsInfo.DeclarationCode, _onTheFlyDocsInfo.Language, cancellationToken).ConfigureAwait(false);
+            var (responseString, responseStatus) = await copilotService.GetOnTheFlyDocsAsync(_onTheFlyDocsInfo.SymbolSignature, _onTheFlyDocsInfo.DeclarationCode, _onTheFlyDocsInfo.Language, cancellationToken).ConfigureAwait(false);
             var copilotRequestTime = stopwatch.Elapsed;
 
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            if (response is null || response.Length == 0)
+            if (responseString is null || responseString.Length == 0)
             {
-                SetResultText(EditorFeaturesResources.An_error_occurred_while_generating_documentation_for_this_code);
-                CurrentState = OnTheFlyDocsState.Finished;
-                Logger.Log(FunctionId.Copilot_On_The_Fly_Docs_Error_Displayed, KeyValueLogMessage.Create(m =>
+                // If the responseStatus is 8, then that means the quota has been exceeded.
+                if (responseStatus == 8)
                 {
-                    m["ElapsedTime"] = copilotRequestTime;
-                }, LogLevel.Information));
+                    this.PlanUpgradeRequested += (_, _) =>
+                    {
+                        // GUID and command ID from
+                        // https://dev.azure.com/devdiv/DevDiv/_wiki/wikis/DevDiv.wiki/45121/Free-SKU-Handling-Guidance-and-Recommendations
+                        var uiShell = _serviceProvider.GetServiceOnMainThread<SVsUIShell, IVsUIShell>();
+                        uiShell.PostExecCommand(
+                            new Guid("39B0DEDE-D931-4A92-9AA2-3447BC4998DC"),
+                            0x3901,
+                            nCmdexecopt: 0,
+                            pvaIn: null);
+
+                        // Telemetry to track when users reach the quota of the Copilot Free plan.
+                        var telemetryEvent = new OperationEvent(
+                            "vs/copilot/showcopilotfreestatus",
+                            TelemetryResult.Success);
+                        telemetryEvent.Properties["vs.copilot.source"] = "CSharpOnTheFlyDocs";
+                        TelemetryService.DefaultSession.PostEvent(telemetryEvent);
+
+                        _asyncQuickInfoSession.DismissAsync();
```

Comment:
should we dismiss before the telemetry?  will this just stay up while the telemetry takes as long as it needs to post?